### ISSUE TITLE
Maximize image on click

### ISSUE THREAD

**Issue by rinigus** [mirukana/mirage#11]
When clicking on image, maximize it instead of opening in the external application (browser)

**Comment by mirukana** [comment]
The image viewer has been added to the dev branch.

For navigating back, we can't use a swipable drawer since that would conflict with flicking the image, but there's a X button at the bottom. Other than that, these controls are available:

- Left click on the image (mouse only): expand to window size if the image's origin size is smaller than the window,  else expand to original size
- Tap on the image (touch screen/pen only): reveal the info and button bars when auto-hidden (bars will auto-hide only when they overlap with a big enough displayed image)
- Any mouse movement: reveal auto-hidden bars 
- Double click on the image: toggle fullscreen 
- Middle click anywhere: open externally
- Right click anywhere: close the viewer, back to chat
- Drag when displayed image is bigger than window to pan
- Wheel to pan up/down, hold shift or alt to pan left/right 
- Ctrl+wheel to control zoom
- Other available keyboard shortcuts are listed [here](https://github.com/mirukana/mirage/blob/dev/src/backend/user_files.py#L385)

I'd like to add pinching gestures to rotate and zoom using [PinchHandler](https://doc.qt.io/qt-5/qml-qtquick-pinchhandler.html), but I have no way to test this without a touch screen, would you be willing to contribute?

**Comment by rinigus** [comment]
Sure, I will be glad to help. Just let me know when it is ready for testing and which branch I should pull in.

**Comment by mirukana** [comment]
It should be ready so you can pull [dev](https://github.com/mirukana/mirage/tree/dev), thanks.

**Comment by rinigus** [comment]
I'll be able to test tonight.

**Comment by rinigus** [comment]
From testing:

- on mobile, tap opens file in a separate application, as according to the configured image://png preferences. Example output:

```
 20:32:19 | Done: file:///home/nemo/.var/app/io.github.mirukana.mirage/cache/mirage/downloads/jameshjacksonjr.modular.im/Screenshot_20200721_010_6a4be624a682611b8a14c05014847e253d7fa678.png
```

- on desktop, I think all works as it should. click with mouse, double click. All very nice.

No idea why it is refusing to do so on mobile. The both were tested using flatpak versions.

**Comment by rinigus** [comment]
PS: used version 87ad3ccab1e9e5ba31e3258c25ec4b7e84ecbdf6

**Comment by mirukana** [comment]
Maybe the middle click handler is getting triggered accidentally? Try adding `print("left", window.settings.media.openExternallyOnClick)` [here](https://github.com/mirukana/mirage/blob/87ad3/src/gui/Pages/Chat/Timeline/EventImage.qml#L94) and `print("middle", window.settings.media.openExternallyOnClick)` [here](https://github.com/mirukana/mirage/blob/87ad3/src/gui/Pages/Chat/Timeline/EventImage.qml#L104).

**Comment by rinigus** [comment]
Yes, you get `~ 10:13:33 | middle false` on click

**Comment by mirukana** [comment]
What happens if you add `acceptedPointerTypes: PointerDevice.GenericPointer |  PointerDevice.Pen` to the middle click `TapHandler`?

**Comment by rinigus** [comment]
yes, was just testing the similar one:  `acceptedDevices: PointerDevice.Mouse | PointerDevice.Stylus`. the one I tested, filtered it out. and shows picture zoomed in.

cannot pan though, as pointer events are processed by panel under and while I try to pan, messages move up-down there

**Comment by mirukana** [comment]
When you said "tap opens file in a separate application", did you mean tap on the timeline thumbnail, or tap on the image in the viewer itself?

**Comment by rinigus** [comment]
in the timeline thumbnail. The fix with acceptedDevices: PointerDevice.Mouse | PointerDevice.Stylus worked, as I expect your proposed fix would work as well.

Issue now is in inability to pan&zoom of the image when in image viewer. This is due to the mouse/tap events being handled by timeline (that's probably the term which you looked for)

**Comment by mirukana** [comment]
As a possible workaround, can you try adding `enabled: ! window.anyPopup` to the [UI component](https://github.com/mirukana/mirage/blob/dev/src/gui/UI.qml#L12)?

**Comment by rinigus** [comment]
I will test it tonight

**Comment by rinigus** [comment]
It did help in terms of not moving the main pane, but not for panning and zooming. Occasionally, after some panning gestures were tried, an image was opened externally again, as before from timeline.
